### PR TITLE
Blocked page access to unauthenticated

### PR DESCRIPTION
**Pull request by zbarnz**
### Context
If a user is not logged in, they should not be able to visit pages requiring authentication.

### Implementation
Pages requiring authentication will now redirect users who are not logged in to the login page

### Merge Checklist
- [x] Correct Target Branch
- [x] Pre-Review Diff Check
- [x] PR Description
- [x] Add Reviewers and Assignees
- [ ] Review Complete
- [ ] Rebase
- [ ] Rebase Approved

### CODE REVIEW

File: www/pages/_app.js [hunk 1]
```diff
@@ -9,6 +9,8 @@ import Layout   from '../components/Layout';
 
 import '../styles/globals.css';
 
+const selectUser = state => state.user;
+
```
Comment (thudsonbu) [inline www/pages/_app.js:13]: I think this is a leftover.

File: www/components/Layout.js [hunk 2]
```diff
@@ -13,19 +13,34 @@ const pagesWithoutDrawer = new Set([
...
   const showDrawer = !pagesWithoutDrawer.has( page );
+  const blockPage = pagesNeedingAuth.has( page );
 
   useEffect( () => {
     document.documentElement.setAttribute(
       'data-theme', storeTheme.theme.theme
     );
+
+    if ( !user._id && blockPage ) {
+      router.push('/login');
```
Comment (thudsonbu) [inline www/components/Layout.js:42]: I was having that issue with the router taking a while to start. Does this work on a hard reload?

Comment (zbarnz) [reply www/components/Layout.js:42]: @thudsonbu You said adding it as a dependency in the use effect will fix that right?

Comment (thudsonbu) [reply www/components/Layout.js:42]: Yeah, you can fix it by adding `router` as a dependency and only use the `router.push` if the router is fully initialized. You might want to experiment with how to check that the router is ready.

Comment (thudsonbu) [reply www/components/Layout.js:42]: I think there is a better solution for this but this should be good for now.

**Review by thudsonbu — APPROVED** [review]

**Pull request merged**